### PR TITLE
[2674] Backfill part time and full time study mode for trainees based…

### PR DESCRIPTION
**Pull request by mooktakim**
… on the course

### Context

https://trello.com/c/fvvy4PyQ/2674-backfill-part-time-and-full-time-study-mode-for-trainees-based-on-the-course

### Changes proposed in this pull request

* Data migration to find all `Trainee` with `course_code` and `study_mode` = `nil`, set `study_mode` to `course.study_mode`

### CODE REVIEW

**Comment by mooktakim** [comment]
[Backfill part time and full time study mode for trainees based on the course](https://trello.com/c/fvvy4PyQ/2674-backfill-part-time-and-full-time-study-mode-for-trainees-based-on-the-course)

**Comment by edwardhorsford** [comment]
@mooktakim does this exclude courses that are dual study mode?

**Comment by mooktakim** [comment]
> @mooktakim does this exclude courses that are dual study mode?

Yes, dual is left blank

File: db/data/20210922135515_backfill_study_mode.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class BackfillStudyMode < ActiveRecord::Migration[6.1]
+  def up
+    Trainee.where(study_mode: nil).where.not(course_code: nil).find_each do |trainee|
```
Comment (jhenderson) [inline db/data/20210922135515_backfill_study_mode.rb:5]:
```suggestion
Trainee.includes(:published_course).where(study_mode: nil).where.not(course_code: nil).find_each do |trainee|
  if (course = trainee.published_course)
    study_mode = course_study_mode_if_valid(course)
    trainee.update!(study_mode: study_mode) if study_mode
  end
end
```

**Review by jhenderson — APPROVED** [review]

**Pull request merged**